### ISSUE TITLE
IVGMM with user-supplied weighting matrix seems to only affect standard errors

### ISSUE THREAD

**Issue by Tututuu** [bashtage/linearmodels#314]
Hello!

In my understanding, the `initial_weight` parameter in `linearmodels.iv.model.IVGMM.fit` should be the first stage weighting matrix used to get an initial parameter estimate &beta;<sub>1 </sub>. After that, we use this &beta;<sub>1 </sub> to generate an estimate for the covariance matrix of the moment conditions &Omega; , and use the inverse of this to finally estimate the &beta;. Therefore, if we change the `initial_weight` parameter, it will naturally change the first stage estimate, and in turn the &Omega; , and finally the final estimate. It seems the `winitial` option in Stata `gmm` command works in this way.

But I just conducted a simple experiment, in which I set `iter_limit=1` to directly see the estimate of the first stage:

```python
import numpy as np
from statsmodels.api import add_constant
from linearmodels.datasets import mroz
print(mroz.DESCR)
data = mroz.load()
data = data.dropna()
data = add_constant(data, has_constant='add')

IVGMM(np.log(data.wage), data[['const']], data.educ, data.fatheduc).fit(iter_limit=1, initial_weight = np.array([[1,2], [2,2]]))
```
which gives me 
![image](https://user-images.githubusercontent.com/36356399/106818106-cc232800-66b2-11eb-9204-46269786e7c3.png)

But when I change the `initial_weight`:
```python
IVGMM(np.log(data.wage), data[['const']], data.educ, data.fatheduc).fit(iter_limit=1, initial_weight = np.array([[1,2], [0.2,0.1]]))
```
which gives me 
![image](https://user-images.githubusercontent.com/36356399/106818227-f70d7c00-66b2-11eb-83dc-30bc8e4996da.png)

The two parameter estimates are the same. The `initial_weight` parameter does not worked quite as I expected. Can you explain a bit why this happen, and how should I do if I want to use user-supplied weighting matrix in the first stage?

Thank you!

**Comment by bashtage** [comment]
If your model is just identified then weights don't matter since the moment conditions are exactly 0.

**Comment by bashtage** [comment]
Definitely a bug.  Should be fixed in #315

**Comment by Tututuu** [comment]
Sorry, my example was inappropriate. But as you just mentioned, the estimates with user-supplied weighting matrices seem to be incorrect. I found this when I was trying to play with the weighting matrix using the data from Acemoglu, Johnson, and Robinson (2001) paper with an over-identified variant of their model. The results using `linearmodels` is different with those using Stata and my manual calculation. 

Finally, thank you a lot for your package!
